### PR TITLE
Use IEC 61499 Coord. System for Conn Bendpoints and Sizes

### PR DESCRIPTION
**Pull request by azoitl**
In order to minimize noise when saving on different screens we now store the bendpoints of connections and element sices in IEC 61499 coordinate system in the model. This should reduce noise in the change history.

### CODE REVIEW

File: plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/properties/GroupPropertySection.java [hunk 1]
```diff
@@ -67,8 +68,8 @@ public void refresh() {
 			commandStack = null;
 			nameText.setText(getType().getName() != null ? getType().getName() : ""); //$NON-NLS-1$
 			commentText.setText(getType().getComment() != null ? getType().getComment() : ""); //$NON-NLS-1$
-			heightText.setText(Integer.toString(getType().getHeight()));
-			widthText.setText(Integer.toString(getType().getWidth()));
+			heightText.setText(Integer.toString(CoordinateConverter.INSTANCE.iec61499ToScreen(getType().getHeight())));
+			widthText.setText(Integer.toString(CoordinateConverter.INSTANCE.iec61499ToScreen(getType().getWidth())));
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/properties/GroupPropertySection.java:72]: Would it make sense for the user to enter IEC 61499 coordinates now?

File: plugins/org.eclipse.fordiac.ide.model/model/lib.ecore [hunk 2]
```diff
@@ -1182,24 +1174,24 @@
...
-        <details key="body" value="return getWidth();"/>
+        <details key="body" value="return org.eclipse.fordiac.ide.model.CoordinateConverter.INSTANCE.iec61499ToScreen(getWidth());"/>
       </eAnnotations>
     </eOperations>
     <eOperations name="getVisibleHeight" eType="ecore:EDataType http://www.eclipse.org/emf/2003/XMLType#//Int">
       <eAnnotations source="http://www.eclipse.org/emf/2002/GenModel">
-        <details key="body" value="return getHeight();"/>
+        <details key="body" value="return org.eclipse.fordiac.ide.model.CoordinateConverter.INSTANCE.iec61499ToScreen(getHeight());"/>
       </eAnnotations>
     </eOperations>
```
Comment (mx990) [inline plugins/org.eclipse.fordiac.ide.model/model/lib.ecore:1196]: How about adding operations to get the *screen* height and width of an FB network element and then add derived attributes to cache the converted values in a group, subapp, or comment (reset on setHeight/setWidth)?

**Review by mx990 — CHANGES_REQUESTED** [review]
I believe you also need to update the calls to `getHeight()` and `getWidth()` in `FBNetworkAnnotations.validateCollisions(...)` to convert to screen coordinates.

**Comment by azoitl** [comment]
> I believe you also need to update the calls to `getHeight()` and `getWidth()` in `FBNetworkAnnotations.validateCollisions(...)` to convert to screen coordinates.

There we are using `getVisibleWidth()` and `getVisibleHeight()` which are performing the translation.

File: plugins/org.eclipse.fordiac.ide.model/model/lib.ecore (hunk 2 above)
Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.model/model/lib.ecore:1196]: That sounds interesting for an subsequent change. I first want to make sure that I haven't broken anything major.

File: plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/properties/GroupPropertySection.java (hunk 1 above)
Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/properties/GroupPropertySection.java:72]: We thought about that. Currently the users sees the ruler and the grid in screen coordinates. Therefore we thought it makes more sense to have them also in the property sheet in screen coordinates.

**Comment by mx990** [comment]
> > I believe you also need to update the calls to `getHeight()` and `getWidth()` in `FBNetworkAnnotations.validateCollisions(...)` to convert to screen coordinates.
> 
> There we are using `getVisibleWidth()` and `getVisibleHeight()` which are performing the translation.

Please have another look at https://github.com/eclipse-4diac/4diac-ide/blob/5bb8160517f72170f4de58274352bf8fe9273efd/plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/libraryElement/impl/FBNetworkAnnotations.java#L97

**Comment by azoitl** [comment]
> > > I believe you also need to update the calls to `getHeight()` and `getWidth()` in `FBNetworkAnnotations.validateCollisions(...)` to convert to screen coordinates.
> > 
> > 
> > There we are using `getVisibleWidth()` and `getVisibleHeight()` which are performing the translation.
> 
> Please have another look at
> 
> https://github.com/eclipse-4diac/4diac-ide/blob/5bb8160517f72170f4de58274352bf8fe9273efd/plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/libraryElement/impl/FBNetworkAnnotations.java#L97

Damn. Missed that. Fix it in a minute.

**Review by mx990 — APPROVED** [review]

**Pull request merged**